src/rp2_common/hardware_pio/include/hardware/pio.h

```diff
@@ -708,6 +708,12 @@ static inline uint pio_get_gpio_base(PIO pio) {
 #endif
 }
 
+static inline void check_pio_pin_mask64(__unused PIO pio, __unused uint64_t pinmask) {
+    // check no pins are set in the mask which are incompatible with the pio
+    valid_params_if(HARDWARE_PIO, (pio_get_gpio_base(pio) == 0 || ((pinmask & ((1ull << pio_get_gpio_base(pio)) - 1)) == 0)) &&
```

Comment:
In `pio.c` the use of `pio_get_gpio_base` is always guarded by `#if PICO_PIO_USE_GPIO_BASE` - does something similar need to be done here too?

Reply:
Also, with all the nested brackets and both an `||` and and `&&` being used here, it's very hard to tell what it's doing!
I may be _totally_ wrong (bit-manipulation isn't my strong point), but would something like this do the trick?
```c
#if PICO_PIO_USE_GPIO_BASE
    valid_params_if(HARDWARE_PIO, pinmask & ~(((1ull << 32) - 1) << pio_get_gpio_base(pio)) == 0);
#else
    valid_params_if(HARDWARE_PIO, pinmask & ~((1ull << 32) - 1) == 0);
#endif
```


I started with
```
    uint64_t allowed_pins = (1ull << 32) - 1;
#if PICO_PIO_USE_GPIO_BASE
    allowed_pins <<= pio_get_gpio_base(pio);
#endif
    valid_params_if(HARDWARE_PIO, pinmask & ~allowed_pins == 0);
```
and then "inlined" the various parts.

Reply:
yes that does look simpler

Reply:
I don't want variables in here because valid_params_if compiles to nothing by default. But I think I've more or less fixed as suggested.